update-extension-dependencies.sh

```diff
@@ -32,7 +46,7 @@ echo ''
 # get all "artifact-id" values from the generated json file
 # pipefail is switched off briefly so that a better error can be logged when nothing is found
 set +o pipefail
-ARTIFACT_IDS=`grep '^    "artifact"' devtools/bom-descriptor-json/target/*.json | grep -Po '(?<=io.quarkus:)(?!quarkus-bom)[^:]+' | sort`
+ARTIFACT_IDS=`grep '^    "artifact"' devtools/bom-descriptor-json/target/*.json | grep -Eo 'quarkus-[a-z0-9-]+' | sort`
```

Comment:
Note: The `quarkus-bom` exclusion is not required anymore since the recent descriptor change.